spec/dfn2toml.py

```diff
@@ -0,0 +1,71 @@
+import argparse
+from collections.abc import Mapping
+from pathlib import Path
+from typing import Any
+
+from flopy.utils import import_optional_dependency
+
+from flopy4.dfn import Dfn
+
+SPEC_PATH = Path(__file__).parent
+DFN_PATH = SPEC_PATH / "dfn"
+TOML_PATH = SPEC_PATH / "toml"
+
+
+class Shim:
+    @staticmethod
+    def _attach_children(d: Any):
+        if isinstance(d, Mapping):
+            if "children" in d:
+                for n, c in d["children"].items():
+                    d[n] = Shim._attach_children(c)
+                del d["children"]
+            d = {k: Shim._attach_children(v) for k, v in d.items()}
+        return d
+
+    @staticmethod
+    def _drop_none(d: Any):
+        if isinstance(d, Mapping):
+            return {
+                k: Shim._drop_none(v)
+                for k, v in d.items()
+                if (v or isinstance(v, bool))
+            }
+        else:
+            return d
```

Comment:
I did some small tests with this method, there might be some bugs with it. For example:

```py
Shim._drop_none({"hey": 0}) # returns {}
Shim._drop_none({"hey": ""}) # return {}
```

Reply:
good catch, thanks